_examples/reddit/reddit.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gocolly/colly"
+)
+
+type item struct {
+	StoryURL  string
+	Source    string
+	comments  string
+	CrawledAt time.Time
+	Comments  string
+	Title     string
+}
+
+func main() {
+	stories := []item{}
+	// Instantiate default collector
+	c := colly.NewCollector(
+		// Visit only domains: reddit.com
+		colly.AllowedDomains("www.reddit.com"),
+	)
+
+	// On every a element which has .top-matter attribute call callback
+	// This class is unique to the div that holds all information about a story
+	c.OnHTML(".top-matter", func(e *colly.HTMLElement) {
+		temp := item{}
+		temp.StoryURL = e.ChildAttr("a[data-event-action=title]", "href")
+		temp.Source = "https://www.reddit.com/r/programming/"
+		temp.Title = e.ChildText("a[data-event-action=title]")
+		temp.Comments = e.ChildAttr("a[data-event-action=comments]", "href")
+		temp.CrawledAt = time.Now()
+		stories = append(stories, temp)
+	})
+
+	// On every span tag with the class next-button
+	c.OnHTML("span.next-button", func(h *colly.HTMLElement) {
+		t := h.ChildAttr("a", "href")
+		c.Visit(t)
+	})
+
+	// Set max Parallelism and introduce a Random Delay
+	c.Limit(&colly.LimitRule{
+		Parallelism: 2,
```

Comment:
Parallelism has no effect if the application is single threaded. Use goroutines or `c.Async = true` to achieve parallel requests.